### DISCUSSION TITLE
Blazor Server Sample

### DISCUSSION THREAD

**Discussion by TanguyIngels** [MarimerLLC/csla#2520] [Questions] [answered]
Hi,

Just wondering if I've done something wrong in.  Downloaded the ProjectTracker sample.  Like explained set up the 2 start projects AppServerCore & Blazor.Server.

If hit the login button:
System.InvalidOperationException: Cannot provide a value for property 'userService' on type 'ProjectTracker.Ui.Blazor.Pages.Account.Login'. There is no registered service of type 'Csla.Blazor.Client.Authentication.CslaUserService'.
When I do F5, login screen appears.

If I exclude the login & logout page from the UI.Blazor project.  Then everything works fine.

Thanks for clarifying,
Tanguy

**Comment by rockfordlhotka** [comment, marked as answer]
Blazor changed the way some of the authn/authz types work from .NET 3.1 to .NET 5 and 6. The ProjectTracker sample (I think) is still built to target the .NET 3.1 types.

**Reply by TanguyIngels** [reply]
Indeed, I get the idea. So I will be able to create my own project :-).  

Tried to follow the upgrade like on https://docs.microsoft.com/en-us/aspnet/core/migration/31-to-50?view=aspnetcore-5.0&tabs=visual-studio#update-blazor-webassembly-and-blazor-server-projects, but the issue still remains. But I've work-around, so I'm good.
